Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -83,8 +85,12 @@ bool ProcessThrottler::addActivity(Activity& activity)
 void ProcessThrottler::removeActivity(Activity& activity)
 {
     ASSERT(isMainRunLoop());
-    if (activity.isForeground())
+    if (activity.isForeground()) {
         m_foregroundActivities.remove(&activity);
+        // If this was the last foreground activity keep holding the foreground assertion for a while
+        if (!m_foregroundActivities.size() && m_shouldDelaySuspension)
+            holdTemporaryForegroundAssertion();
```

Comment:
Why do we need to hold a *foreground* assertion for a background tab. This seems a bit confusing. Why can't we hold a Background assertion?